internal/config/types.go

```diff
@@ -123,17 +125,63 @@ func (c *Config) HasSecret(secret string) bool {
 type Environment = map[string]string
 
 type Python struct {
-	Version        string `toml:"version" json:"version"`
+	Version        string `toml:"version,omitempty" json:"version"`
 	PackageFile    string `toml:"package_file,omitempty" json:"packageFile"`
 	PackageManager string `toml:"package_manager,omitempty" json:"packageManager"`
 }
 
+func (p *Python) FillDefaults(
+	pythonInterpreter *interpreters.PythonInterpreter,
```

Comment:
You shouldn't need to use pointers on interfaces,
removing `*` from here (`pythonInterpreter interpreters.PythonInterpreter`) will make things easier below:
```
pythonInterpreter.IsPythonExecutableValid()
```

Reply:
The code and tests have been reworked. Thanks!